openedx/core/djangoapps/user_api/preferences/api.py

```diff
@@ -110,7 +112,8 @@ def update_user_preferences(requesting_user, update, username=None):
         PreferenceUpdateError: the operation failed when performing the update.
         UserAPIInternalError: the operation failed due to an unexpected error.
     """
-    existing_user = _get_user(requesting_user, username)
+    if not existing_user:
```

Comment:
If we separate them, I think it makes sense to fully separate them.  The way it's written above, it looks like the user is only authorized if a user object was passed in:

    if isinstance(user, basestring):
        user = _get_user(requesting_user, user)
    _authorize(requesting_user, user)

That way it's clear without digging through the code that the authorization happens regardless of how the user is passed in.

As a side note, use `isinstance(user, basestring)`, so it works properly if a `unicode` string is passed in too.